service/product/service.go

```diff
@@ -56,3 +60,149 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
 	log.Printf("Product %s created successfully", p.Name)
 	helper.BaseResponse(w, response, http.StatusOK)
 }
+
+func DeleteProduct(w http.ResponseWriter, r *http.Request) {
+	var id viewmodels.ItemIdRequest
+	err := json.NewDecoder(r.Body).Decode(&id)
+	if err != nil {
+		log.Println("Failed to decode request body: ", err)
+		response, _ := json.Marshal(viewmodels.BaseResponse{
+			StatusCode: statusCodes.FailedToDecodeRequestBody,
+			Message:    statusCodes.StatusCodes[statusCodes.FailedToDecodeRequestBody],
+		})
+		helper.BaseResponse(w, response, http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	err = id.ValidateItemIdRequest()
+
+	if err != nil {
+		log.Println("Failed to validate request body: ", err)
+		response, _ := json.Marshal(viewmodels.BaseResponse{
+			StatusCode: statusCodes.FailedToValidateItemIdRequest,
+			Message:    statusCodes.StatusCodes[statusCodes.FailedToValidateItemIdRequest],
+		})
+		helper.BaseResponse(w, response, http.StatusBadRequest)
+		return
+	}
+
+	claims := helper.CheckToken(&w, r)
```

Comment:
this can be above because if token is not ok you should not proceede, this can be adjusted everywhere